### PR TITLE
planner: add missing distinct flag for Apply convert to join

### PR DESCRIPTION
**Pull request by wshwsh12**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/26958 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Check List 

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
Please add a release note, or a 'None' if it is not needed.
```

### CODE REVIEW

**Review by winoros — APPROVED** [review]

File: planner/core/rule_decorrelate.go [hunk 1]
```diff
@@ -192,7 +192,7 @@ func (s *decorrelateSolver) optimize(ctx context.Context, p LogicalPlan) (Logica
 					switch expr := aggFunc.Args[0].(type) {
 					case *expression.Column:
 						if idx := apply.schema.ColumnIndex(expr); idx != -1 {
-							desc, err := aggregation.NewAggFuncDesc(agg.ctx, agg.AggFuncs[i].Name, []expression.Expression{apply.schema.Columns[idx]}, false)
+							desc, err := aggregation.NewAggFuncDesc(agg.ctx, agg.AggFuncs[i].Name, []expression.Expression{apply.schema.Columns[idx]}, agg.AggFuncs[i].HasDistinct)
```
Comment (wjhuang2016) [inline planner/core/rule_decorrelate.go:195]: I found that many usages use `false`, does they need to change too?

Comment (wshwsh12) [reply planner/core/rule_decorrelate.go:195]: Need consider situation case by case.  In this case, it is an aggFuncs rewrite logic, so we should keep the property.

Comment (wshwsh12) [reply planner/core/rule_decorrelate.go:195]: The AggFuncs in other usage is only FirstRow,Min/Max,Count.   FirstRow,Min/Max don't need to consider distinct.
There are only four usages of the function by Count.  It seems that their usage is correct.

**Review by lzmhhh123 — APPROVED** [review]

**Pull request merged**